dashboard/src/actions/overviewActions.js

```diff
@@ -246,3 +257,55 @@ export const setLoadingDoneFlag = () => async (dispatch, getState) => {
     }
   }, DASHBOARD_LOAD_DELAY_MS);
 };
+
+const filterDatasetType = (type) => (getState) => {
+  if (type === "newRuns") {
+    return getState().overview.initNewRuns;
+  }
+  return getState().overview.savedRuns;
+};
+
+const updateDatasetType = (data, type) => (dispatch) => {
+  if (type === "newRuns") {
+    dispatch({
+      type: TYPES.INIT_NEW_RUNS,
+      payload: data,
+    });
+  } else {
+    dispatch({
+      type: TYPES.SAVED_RUNS,
+      payload: data,
+    });
+  }
+};
+export const editMetadata =
+  (value, metadata, rId, type) => async (dispatch, getState) => {
+    const data = filterDatasetType(type)(getState);
+
+    const rIndex = data.findIndex((item) => item.resource_id === rId);
+    data[rIndex][metadata] = value;
+    data[rIndex]["isDirty"] = true;
```

Comment:
It's nice for constants like `"isDirty`, `"error"`, `"success"`, `"name_validated"`, `"name_errorMsg"` and such to be defined centrally as constants. That way, if they are spelled incorrectly somewhere the language will complain rather than failing in subtle ways. (For example, because you're looking for `"success"` but some place in the code set it to `"sucess"`.)

Reply:
Updated for few properties.